### PR TITLE
Switch to go 1.23

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Update go.mod to require go 1.23.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Require at least go 1.23 to build this project.
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@kubernetes-sigs/release-engineering PTAL

File: go.mod [hunk 1]
```diff
@@ -16,28 +16,27 @@ require (
...
+	github.com/pierrec/lz4/v4 v4.1.2 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
-	github.com/ulikunitz/xz v0.5.12 // indirect
+	github.com/ulikunitz/xz v0.5.9 // indirect
 	github.com/xi2/xz v0.0.0-20171230120015-48954b6210f8 // indirect
-	golang.org/x/mod v0.17.0 // indirect
-	golang.org/x/sync v0.7.0 // indirect
-	golang.org/x/sys v0.20.0 // indirect
-	golang.org/x/text v0.15.0 // indirect
```
Comment (ameukam) [inline go.mod:40]: any reason why we downgrade ?

Comment (saschagrunert) [reply go.mod]: No, fixed that.

**Comment by saschagrunert** [comment]
@kubernetes-sigs/release-engineering PTAL

**Pull request merged**